kratos/tests/cpp_tests/geometries/test_quadrilateral_2d_4.cpp

```diff
@@ -173,5 +188,57 @@ namespace Testing
         TestAllShapeFunctionsLocalGradients(*geom);
     }
 
+    KRATOS_TEST_CASE_IN_SUITE(Quadrilateral2D4LumpingFactorsRegularShape, KratosCoreGeometriesFastSuite) {
+        auto geom = GenerateRightQuadrilateral2D4<Node<3>>();
+
+        Vector lumping_factors(4);
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::ROW_SUM);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[1], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[2], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[3], 0.25, TOLERANCE);
+
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::DIAGONAL_SCALING);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[1], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[2], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[3], 0.25, TOLERANCE);
+
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::QUADRATURE_ON_NODES);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[1], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[2], 0.25, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[3], 0.25, TOLERANCE);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(Quadrilateral2D4LumpingFactorsTrapezoidalShape, KratosCoreGeometriesFastSuite) {
+        auto geom = GenerateTrapezoidalQuadrilateral2D4<Node<3>>();
+
+        Vector lumping_factors(4);
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::ROW_SUM);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.3, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[1], 0.3, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[2], 0.2, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[3], 0.2, TOLERANCE);
+
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::DIAGONAL_SCALING);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.325, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[1], 0.325, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[2], 0.175, TOLERANCE);
+        KRATOS_CHECK_NEAR(lumping_factors[3], 0.175, TOLERANCE);
+
+        geom->LumpingFactors(lumping_factors, Geometry<Node<3>>::LumpingMethods::QUADRATURE_ON_NODES);
+
+        KRATOS_CHECK_NEAR(lumping_factors[0], 0.4, TOLERANCE);
```

Comment:
apparently macro substitution is quite slow, and it would be better to compare vectors to vectors.

@philbucher @roigcarlo could u point once again to the way of doing it in tests?

Reply:
yep also saw that
Afaik the macro unrolling is low (@roigcarlo please correct me if I am wrong)
BTW @roigcarlo do we really need the macros here or could we use functions instead? => I will create an issue

for this particular case with our current code I suggest the following: 
~~~c
const std::vector<double> expected_results{ 0.4, 0.4, 0.1, 0.1 };
for (int i=0; i<4; ++i) {
    KRATOS_CHECK_NEAR(lumping_factors[i], expected_results[i], TOLERANCE);
}
~~~